quickwit/quickwit-indexing/src/source/file_source.rs

```diff
@@ -17,45 +17,36 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program. If not, see <http://www.gnu.org/licenses/>.
 
-use std::borrow::Borrow;
-use std::ffi::OsStr;
-use std::path::Path;
+use std::fmt;
 use std::time::Duration;
-use std::{fmt, io};
 
-use anyhow::Context;
-use async_compression::tokio::bufread::GzipDecoder;
 use async_trait::async_trait;
-use bytes::Bytes;
 use quickwit_actors::{ActorExitStatus, Mailbox};
-use quickwit_common::uri::Uri;
 use quickwit_config::FileSourceParams;
-use quickwit_metastore::checkpoint::PartitionId;
+use quickwit_metastore::checkpoint::{PartitionId, SourceCheckpoint};
 use quickwit_proto::metastore::SourceType;
-use quickwit_proto::types::{Position, SourceId};
-use serde::Serialize;
-use tokio::io::{AsyncBufReadExt, AsyncRead, AsyncReadExt, BufReader};
-use tracing::info;
+use quickwit_proto::types::SourceId;
 
-use super::BatchBuilder;
+use super::doc_file_reader::ObjectUriBatchReader;
+#[cfg(feature = "sqs")]
+use super::queue_sources::coordinator::QueueCoordinator;
 use crate::actors::DocProcessor;
 use crate::source::{Source, SourceContext, SourceRuntime, TypedSourceFactory};
 
-/// Number of bytes after which a new batch is cut.
-pub(crate) const BATCH_NUM_BYTES_LIMIT: u64 = 500_000u64;
-
-#[derive(Default, Clone, Debug, Eq, PartialEq, Serialize)]
-pub struct FileSourceCounters {
-    pub previous_offset: u64,
-    pub current_offset: u64,
-    pub num_lines_processed: u64,
+enum FileSourceInner {
```

Comment:
`FileSourceInner` is not a good name. 

We use the `Inner` suffix for `Pimpl` like patterns for instance, or when we need to wrap struct with a `Mutex`.